### PR TITLE
Made ClusteringServiceConfiguration return a copy of Properties

### PR DESCRIPTION
**Pull request by saurabhagas**

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ public Timeouts getTimeouts() {
    * @return the connection {@code Properties}
    */
   public Properties getProperties() {
-    return properties;
+    return new Properties(properties);
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: This does not do what you believe it does :wink:

Also, you need to take a copy when assigning it to the field during construction to protect against mutation from the object you received.

Comment (saurabhagas) [reply clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: Done

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public ClusteringServiceConfiguration(URI clusterUri, Timeouts timeouts, boolean
     this.autoCreate = autoCreate;
     this.serverConfiguration = serverConfig;
     this.timeouts = Objects.requireNonNull(timeouts, "Operation timeouts cannot be null");
-    this.properties = Objects.requireNonNull(properties, "Properties cannot be null");
+    this.properties = cloneProperties(Objects.requireNonNull(properties, "Properties cannot be null"));
```
Comment (henri-tremblay) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: You could do `(Properties) Objects.requireNonNull(properties, "Properties cannot be null").clone();` and be done with it.

**Comment by saurabhagas** [comment]
@henri-tremblay, I've done the suggested change.

**Comment by saurabhagas** [comment]
test this please

**Pull request merged**